ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobMetadata.java

```diff
@@ -399,8 +420,72 @@ public boolean equals(Object o) {
       return false;
     }
     CloudBlobMetadata om = (CloudBlobMetadata) o;
-    return (Objects.equals(id, om.id) && Objects.equals(partitionId, om.partitionId) && creationTime == om.creationTime
+    return (Objects.equals(id, om.id) && Objects.equals(partitionId, om.partitionId) && accountId == om.accountId
+        && containerId == om.containerId && size == om.size && creationTime == om.creationTime
         && uploadTime == om.uploadTime && expirationTime == om.expirationTime && deletionTime == om.deletionTime
-        && size == om.size && accountId == om.accountId && containerId == om.containerId);
+        && namingScheme == om.namingScheme && encryptionOrigin == om.encryptionOrigin && Objects.equals(vcrKmsContext,
+        om.vcrKmsContext) && Objects.equals(cryptoAgentFactory, om.cryptoAgentFactory)
+        && encryptedSize == om.encryptedSize);
+  }
+
+  /**
+   * @return a {@link HashMap} of metadata key-value pairs.
+   */
+  public HashMap<String, String> toMap() {
+    ObjectMapper objectMapper = new ObjectMapper();
+    HashMap<String, String> map = new HashMap<>();
+    ObjectNode node = objectMapper.convertValue(this, ObjectNode.class);
+    node.fieldNames().forEachRemaining(fieldName -> {
+      map.put(fieldName, node.get(fieldName).asText());
+    });
+    return map;
```

Comment:
I changed it to convert directly to a map.  It still does json serialization internally, so not as efficient as handcrafting the map.  But I prefer to avoid handcrafting since it's easy to forget to keep it in sync.

Reply:
cool, that `TypeReference` serializer seems useful.